### ISSUE TITLE
Support images in CRDs

### ISSUE THREAD

**Issue by landism** [windmilleng/tilt#1069]
Tilt currently knows how to find images defined in "containers" for stock k8s types, but if a CRD specifies an image (even if it's using standard container syntax), Tilt doesn't know what to do with it.

**Comment by landism** [comment]
This turned out to be trickier than I had realized:

When Tilt applies a k8s config, it injects some labels into any ObjectMeta fields it finds. It then uses these labels to figure out which pods it should show in the UI, and to join pods to Tilt resources.

When a CRD is involved, it doesn't necessarily have an ObjectMeta field, or any other generic way for us to identify which pods it's creating.

As a relatively cheap solution, I'm going to add an option to the Tiltfile's `k8s_resource` that allows the user to specify labels that will be monitored for that resource, regardless of how they were created.

**Comment by landism** [comment]
I've landed #1087. Once #1117 lands, this will be unblocked, but with some rough edges.

I've been testing with [fission](https://github.com/fission/fission).

Rough edges remaining:
- Tiltfile loading doesn't work quite the same way for CRDs, since we don't know in advance where the image labels are defined, and it's non-trivial to remove that need. At the moment, instead of defining a `k8s_yaml`, you'll need to define a `k8s_resource` and give it the yaml and image directly, e.g.:
```
docker_build('gcr.io/windmill-public-containers/servantes/go-env', 'fission')
k8s_resource('go-env', yaml='deploy/fission.yaml', image='gcr.io/windmill-public-containers/servantes/go-env', extra_pod_selectors=[{'environmentName': 'go'}])
```

- You might note above that `k8s_resource` also got an `extra_pod_selectors`. This is to allow Tilt to find the pods for the resource, as mentioned in the previous comment.

- Fission defaults to 3 replicas per pool, and Tilt in general behaves a little weirdly when a deployment has multiple replicas. To get around this, I used `--version 3 --poolsize 1` when creating my Fission environments.

This will hopefully unblock using Tilt with CRDs, and we'll worry about polishing those rough edges as future work based on feedback.
